packages/catalog-client/src/CatalogClient.ts

```diff
@@ -46,9 +46,14 @@ import { DiscoveryApi } from './types/discovery';
  * */
 export class CatalogClient implements CatalogApi {
   private readonly discoveryApi: DiscoveryApi;
+  private readonly fetch: typeof crossFetch;
 
-  constructor(options: { discoveryApi: DiscoveryApi }) {
+  constructor(options: {
+    discoveryApi: DiscoveryApi;
+    fetch?: typeof crossFetch;
```

Comment:
Probably just wanna make this `FetchApi` tbh? Just to make sure we're sticking to mostly passing APIs around and keep migrations smooth

Reply:
Yep, agreed, the ugly part is that this package doesn't depend on core-plugin-api so I have duplicated the types in here - but, fixed the same way as with `DiscoveryApi` now